src/components/inputs/content-switcher/ohri-content-switcher.component.tsx

```diff
@@ -20,31 +20,37 @@ export const OHRIContentSwitcher: React.FC<OHRIFormFieldProps> = ({ question, on
     }
   }, [question]);
 
-  const handleChange = value => {
+  const handleChange = (value) => {
     setFieldValue(question.id, value?.name);
     onChange(question.id, value?.name, setErrors, null);
     question.value = handler?.handleFieldSubmission(question, value?.name, encounterContext);
   };
 
   const selectedIndex = useMemo(
-    () => question.questionOptions.answers.findIndex(option => option.concept == field.value),
+    () => question.questionOptions.answers.findIndex((option) => option.concept == field.value),
     [field.value, question.questionOptions.answers],
   );
 
   useEffect(() => {
-    getConceptNameAndUUID(question.questionOptions.concept).then(conceptTooltip => {
+    getConceptNameAndUUID(question.questionOptions.concept).then((conceptTooltip) => {
       setConceptName(conceptTooltip);
     });
   }, [conceptName, question.questionOptions.concept]);
 
   const isInline = useMemo(() => {
-    if (encounterContext.sessionMode == 'view' || isTrue(question.readonly)) {
-      return isInlineView(question.inlineRendering, layoutType, workspaceLayout);
+    if (
+      encounterContext.sessionMode == 'view' ||
+      encounterContext.sessionMode == 'embedded-view' ||
```

Comment:
To reduce the duplication with this maybe we can introduce union type and use it for all occurrences of this like so; (looking at the rest of the code this might need to be in the types files for reuse)

```
type ReadOnlyMode = 'view' | 'embedded-view'
```

That will shorten the duplication to this 

```
if (encounterContext.sessionMode == ReadOnlyMode) {
  ...
}
```

Reply:
@pirupius I think a better solution would be to use this approach

`if (['view', 'embedded-view'].includes(encounterContext.sessionMode))`
 
the comparison above compares types against variables.

Reply:
@arodidev the main goal is to get rid of the various repetitions and also have type checkings. I made a mistake in my comment but i edited it to  to reflect that. 

```
encounterContext.sessionMode == typeof ReadOnlyMode
```